lib/symbol-index.coffee

```diff
@@ -150,6 +150,8 @@ class SymbolIndex
           matches.push(symbol)
 
   processFile: (fqn) ->
+    if not @keepPath(fqn)
```

Comment:
What would you think about just updating https://github.com/v3ss0n/goto/pull/102/files#diff-93954ac5275f714bda2fb60c4f378d2fR113 to `@processFile filePath if @keepPath filePath`? It would maintain the `@keepPath` checks in the same place during fs tree traversal and avoid duplicate work from calling `@processFile` on line 107.